### PR TITLE
Spring 06

### PR DESCRIPTION
**Pull request by semyonich**

### CODE REVIEW

File: src/main/java/com/cinema/controller/ShoppingCartController.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    public void addMovieSession(@RequestParam Long userId, @RequestParam Long movieSessionId) {
+        User user = userService.findById(userId)
+                .orElseThrow(() -> new DataProcessingException("User don't exist, id=" + userId));
+        MovieSession movieSession = movieSessionService.findById(movieSessionId)
+                .orElseThrow(() -> new DataProcessingException("MovieSession don't exist, id="
+                        + movieSessionId));
+        shoppingCartService.addSession(movieSession, user);
+    }
+
+    //Get by user - GET: /shopping-carts/by-user?userId
```
Comment (Ketch92) [inline src/main/java/com/cinema/controller/ShoppingCartController.java]: Keep it or remove

**Review by Ketch92 — APPROVED** [review]
Well done!

File: src/main/java/com/cinema/controller/ShoppingCartController.java (hunk 1 above)
Comment (semyonich) [reply src/main/java/com/cinema/controller/ShoppingCartController.java]: done

File: src/main/java/com/cinema/dao/OrderDao.java [hunk 2]
```diff
@@ -6,4 +6,6 @@
 
 public interface OrderDao extends AbstractDao<Order> {
     List<Order> getOrdersHistory(User user);
+
+    List<Order> getOrdersHistory(Long userId);
```
Comment (KseniiaMakarova) [inline src/main/java/com/cinema/dao/OrderDao.java]: let's remove this method and use `List<Order> getOrdersHistory(User user);` that we already have

File: src/main/java/com/cinema/service/MovieSessionService.java [hunk 3]
```diff
@@ -12,4 +13,6 @@
     void update(MovieSession movieSession);
 
     void removeById(Long id);
+
+    Optional<MovieSession> findById(Long movieSessionId);
```
Comment (KseniiaMakarova) [inline src/main/java/com/cinema/service/MovieSessionService.java]:
```suggestion
    MovieSession get(Long movieSessionId);
```
let's return Optional from DAO layer, and entity from service layer

File: src/main/java/com/cinema/service/ShoppingCartService.java [hunk 4]
```diff
@@ -12,4 +13,6 @@
     void registerNewShoppingCart(User user);
 
     void clear(ShoppingCart shoppingCart);
+
+    Optional<ShoppingCart> findById(Long id);
```
Comment (KseniiaMakarova) [inline src/main/java/com/cinema/service/ShoppingCartService.java]: same

File: src/main/java/com/cinema/service/UserService.java [hunk 5]
```diff
@@ -7,4 +7,6 @@
     User add(User user);
 
     Optional<User> findByEmail(String email);
+
+    Optional<User> findById(Long id);
```
Comment (KseniiaMakarova) [inline src/main/java/com/cinema/service/UserService.java]: same

File: src/main/java/com/cinema/service/UserService.java [hunk 6]
```diff
@@ -7,4 +7,6 @@
     User add(User user);
 
     Optional<User> findByEmail(String email);
```
Comment (KseniiaMakarova) [inline src/main/java/com/cinema/service/UserService.java:9]: but for this method it is ok to return Optional from service

File: src/main/java/com/cinema/service/OrderService.java [hunk 7]
```diff
@@ -9,4 +9,6 @@
     Order completeOrder(ShoppingCart shoppingCart);
 
     List<Order> getOrdersHistory(User user);
+
+    List<Order> getOrdersHistory(Long userId);
```
Comment (KseniiaMakarova) [inline src/main/java/com/cinema/service/OrderService.java]: let's remove this

File: src/main/java/com/cinema/service/mapper/OrderMapper.java [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+import java.time.format.DateTimeFormatter;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.springframework.stereotype.Component;
+
+@Component
+public class OrderMapper {
+    private static final String DATE_TIME_FORMAT = "dd.MM.yyyy HH:mm";
+
+    public OrderResponseDto toDto(Order order) {
```
Comment (KseniiaMakarova) [inline src/main/java/com/cinema/service/mapper/OrderMapper.java]:
```suggestion
    public OrderResponseDto makeDto(Order order) {
```

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/com/cinema/service/mapper/OrderMapper.java (hunk 8 above)
Comment (semyonich) [reply src/main/java/com/cinema/service/mapper/OrderMapper.java]: done

File: src/main/java/com/cinema/dao/OrderDao.java (hunk 2 above)
Comment (semyonich) [reply src/main/java/com/cinema/dao/OrderDao.java]: done

File: src/main/java/com/cinema/service/MovieSessionService.java (hunk 3 above)
Comment (semyonich) [reply src/main/java/com/cinema/service/MovieSessionService.java]: done

File: src/main/java/com/cinema/service/ShoppingCartService.java (hunk 4 above)
Comment (semyonich) [reply src/main/java/com/cinema/service/ShoppingCartService.java]: done

File: src/main/java/com/cinema/service/UserService.java (hunk 6 above)
Comment (semyonich) [reply src/main/java/com/cinema/service/UserService.java:9]: left as is

File: src/main/java/com/cinema/service/UserService.java (hunk 5 above)
Comment (semyonich) [reply src/main/java/com/cinema/service/UserService.java]: done

File: src/main/java/com/cinema/service/OrderService.java (hunk 7 above)
Comment (semyonich) [reply src/main/java/com/cinema/service/OrderService.java]: removed

File: src/main/java/com/cinema/service/UserService.java (hunk 5 above)
Comment (semyonich) [reply src/main/java/com/cinema/service/UserService.java]: refactored

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**